forte/processors/base/base_processor.py

```diff
@@ -38,11 +39,34 @@ def __init__(self):
         super().__init__()
         self.selector = DummySelector()
 
+    def record(self, record_meta: Dict):
+        r"""Method to add output type record of current processor
+        to Meta attribute record dictionary.
+        """
+        pass
+
+    @classmethod
+    def expected_type(cls) -> Dict:
+        r"""Method to add expected type for current processor input which
+        would be checked before running the processor if enforce_consistency()
+        was called for the pipeline.
+        """
+        return {}
+
     def process(self, input_pack: PackType):
+        if self._check_type_consistency:
+            expected_types = self.expected_type()
+            # check if expected types are in input pack
+            for expected_t in expected_types:
+                if expected_t not in input_pack._meta.record.keys():
+                    raise ExpectedRecordNotFound(
+                        f"The record type {expected_t} is not found in "
+                        f"meta of the provided pack.")
         # Set the component for recording purpose.
         input_pack.set_control_component(self.name)
         self._process(input_pack)
-
+        # add type record of output to meta
```

Comment:
Use full sentences for comments if we can. Here: 
"Add type records of output to the `meta` of the `input_pack`."